jobs/jobs.go

```diff
@@ -0,0 +1,73 @@
+package jobs
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jmoiron/sqlx"
+	"gopkg.in/guregu/null.v3"
+)
+
+type JobCarrer struct {
+	Id               int         `json:"id"`
+	Title            string      `json:"title"`
+	Summary          string      `json:"summary"`
+	Description      string      `json:"description"`
+	ButtonName       string      `json:"button_name"`
+	Qualification    string      `json:"qualification"`
+	EmploymentType   string      `json:"employment_type"`
+	BaseSalary       int         `json:"base_salary"`
+	Experience       string      `json:"experience"`
+	IsActive         bool        `json:"is_active"`
+	Skills           null.String `json:"skills"`
+	TotalOpenings    int         `json:"total_openings"`
+	Responsibilities null.String `json:"responsibilities"`
+}
+
+type JobRepository struct {
+	Db *sqlx.DB
+}
+
+func New(db *sqlx.DB) *JobRepository {
+	return &JobRepository{Db: db}
+}
+
+func (repository *JobRepository) Jobs(c *gin.Context) {
+	var carrersList []JobCarrer
+
+	err := repository.Db.Select(&carrersList, `SELECT id, title, summary, description, button_name, qualification, employment_type, base_salary, experience, is_active, skills, total_openings, responsibilities FROM jobs WHERE is_active = 1 `)
+
+	if err != nil {
+		fmt.Println(err)
+		c.AbortWithStatus(http.StatusInternalServerError)
+		return
+	}
+
+	c.JSON(http.StatusOK, carrersList)
+
+}
+
+func (repository *JobRepository) JobById(c *gin.Context) {
+
+	job := JobCarrer{}
+
+	id, err := strconv.Atoi(c.Param("id"))
+
+	if err != nil {
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	err = repository.Db.Get(&job, `SELECT id, title, summary, description, button_name, qualification, employment_type, base_salary, experience, is_active, skills, total_openings, responsibilities FROM jobs
+										WHERE id = ?
+										AND is_active = 1`, id)
+
+	if err != nil {
+		c.AbortWithStatus(http.StatusInternalServerError)
```

Comment:
What if data is not found for the given id? we should return 404 instead of 500. Also, log error here.